lib/Mojo/Promise.pm

```diff
@@ -18,8 +18,9 @@ sub AWAIT_FAIL         { _settle_await(reject  => @_) }
 sub AWAIT_GET {
   my $self    = shift;
   my @results = @{$self->{results} // []};
-  die $results[0] unless $self->{status} eq 'resolve';
-  return wantarray ? @results : $results[0];
+  return wantarray ? @results : $results[0] if $self->{status} eq 'resolve';
+  die $results[0]                           if ref $results[0] || $results[0] =~ m!\n!;
+  croak $results[0];
```

Comment:
Does this really need `die` and `croak`? Don't like so much logic just to throw an exception. Ternary would probably also look better here.

Reply:
Test::Future::AsyncAwait::Awaitable fails if I use croak and there's a newline in the message:

```
# Subtest: Mojo::Promise conforms to Awaitable API immediate fail
    not ok 5 - AWAIT_GET throws exception

    #   Failed test 'AWAIT_GET throws exception'
    #   at Test/Future/AsyncAwait/Awaitable.pm line 129.
    #          got: 'Oopsie
    #  at Test/Future/AsyncAwait/Awaitable.pm line 128.
    # '
    #     expected: 'Oopsie
    # '
    1..5
```